### PR TITLE
Fix #1383: map most recognized SERVER_ errors to 504 or 502, leave 500 for unrecognized

### LINKED ISSUE #1383: Exceptions from Database (via CQL Driver) incorrectly mapped to HTTP 500, should be 504 (or 502)
HTTP response 500 indicates an Internal Server error (see f.ex https://en.wikipedia.org/wiki/List_of_HTTP_status_codes). There are some failure modes that should be mapped to this code, for example otherwise unhandled exceptions.

But we have many exceptions thrown by CQL driver, caught and re-processed by `ThrowableToErrorMapper`, where we use code 500 as well.
These would much better map to 504 (Gateway Timeout) or sometimes 502 (Bad Gateway).

The reason for changing these is to improve operational support: 500 means something particularly bad happened; whereas 504 generally means problem due to a system Data API depends on (DB or possibly Embedding Gateway).

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Maps most former 500 http failures to 504 or 502, to indicate these are recognized failures with backend DB; 500 remains for unknown/unrecognized exceptions

**Which issue(s) this PR fixes**:
Fixes #1383

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/service/operation/collections/FindCollectionOperationTest.java [hunk 1]
```diff
@@ -3018,15 +3018,15 @@ public void readFailureException() throws UnknownHostException {
           ThrowableToErrorMapper.getMapperWithMessageFunction()
               .apply(failure, failure.getMessage());
       AssertionsForClassTypes.assertThat(error).isNotNull();
-      AssertionsForClassTypes.assertThat(error.message())
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/service/operation/collections/FindCollectionOperationTest.java:3021]: Just reordered to start with more general tests (error code, exception class). And then changed testing for message according to change in `ThrowableToErrorMapper`

File: src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java [hunk 2]
```diff
@@ -88,17 +89,21 @@ private static CommandResult.Error handleDriverException(
       return handleAllNodesFailedException((AllNodesFailedException) throwable, message);
     } else if (throwable instanceof ClosedConnectionException) {
       return ErrorCode.SERVER_CLOSED_CONNECTION
-          .toApiException(message)
-          .getCommandResultError(Response.Status.INTERNAL_SERVER_ERROR);
+          .toApiException("(DriverException/ClosedConnectionException) %s", message)
+          .getCommandResultError(Response.Status.GATEWAY_TIMEOUT);
```
Comment (Hazel-Datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java:93]: Why here is `GATEWAY_TIMEOUT` not `BAD_GATEWAY`?

Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java:93]: Let me double-check; you are right seems odd.

Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java]: @Hazel-Datastax yes, you are right: should be BAD_GATEWAY -- changing. Thank you!

**Review by maheshrajamani — APPROVED** [review]
LGTM

**Review by Hazel-Datastax — APPROVED** [review]

**Review by vkarpov15 — COMMENTED** [review]
LGTM

**Pull request merged**